src/base/bittorrent/session.cpp

```diff
@@ -1567,6 +1568,29 @@ void Session::handleDownloadFinished(const Net::DownloadResult &result)
     }
 }
 
+void Session::updateTorrentStatusReport()
+{
+    m_torrentStatusReport = {};
+    for (const TorrentHandle *torrent : asConst(m_torrents)) {
+        if (torrent->isDownloading())
```

Comment:
If you want the following can be `else if`'d:
isDownloading - isCompleted - isErrored(here)
isPaused - isResumed
isActive - isInactive - isErrored(or here)

As for `isErrored` it can also be inside `if (isPaused())`

Not sure if it's a good idea for `isDownloading - isCompleted` though because people have asked to change what isCompleted means and if that happens this place may be forgotten.

Reply:
Err, this sounds like something that belongs to another PR, I only wrapped the code into a function. Feel free to open one to discuss.